shell/models/provisioning.cattle.io.cluster.js

```diff
@@ -891,7 +891,37 @@ export default class ProvCluster extends SteveModel {
   }
 
   get hasError() {
-    return this.status?.conditions?.some((condition) => condition.error === true);
+    // Before we were just checking for this.status?.conditions?.some((condition) => condition.error === true)
+    // but this is wrong as an error might exist but it might not be meaningful in the context of readiness of a cluster
+    // which is what this 'hasError' is used for.
+    // We now check if there's a ready condition after an error, which helps dictate the readiness of a cluster
+    // Based on the findings in https://github.com/rancher/dashboard/issues/10043
+    if (this.status?.conditions && this.status?.conditions.length) {
+      if (this.status?.conditions.some((c) => c.error === true)) {
```

Comment:
Can we add a comment to say - if any conditions are in error, then we use additional logic, otherwise return quickly as we know there are no errors